tests/OperationProcessor.spec.ts

```diff
@@ -233,11 +250,273 @@ describe('OperationProcessor', async () => {
+    const publicKey2 = Document.getPublicKey(didDocument!, 'key2');
+    expect(publicKey2).toBeDefined();
+    expect(publicKey2!.owner).toBeUndefined(); // if update above went through, the owner would be defined
+  });
+
+  it('should ignore update operation with an invalid signature', async () => {
+    await operationProcessor.processBatch([createOp!]);
+
+    const updatePayload = {
+      didUniqueSuffix,
+      operationNumber: 1,
+      previousOperationHash: createOp!.getOperationHash(),
+      patch: [{
+        op: 'replace',
+        path: '/publicKey/1',
+        value: {
+          id: '#key2',
+          type: 'RsaVerificationKey2018',
+          owner: 'did:sidetree:updateid1',
+          publicKeyPem: process.hrtime() // Some dummy value that's not used.
+        }
+      }]
+    };
+
+    // Generate operation with an invalid key
+    const updateOperation = await OperationGenerator.generateUpdateOperation(updatePayload, '#key1', privateKey);
+    updateOperation.signature = 'InvalidSignature';
+    const updateOperationBuffer = Buffer.from(JSON.stringify(updateOperation));
+    const anchoredUpdateOperation = await addBatchFileOfOneOperationToCas(updateOperationBuffer, cas, 1, 1, 0);
+    await operationProcessor.processBatch([anchoredUpdateOperation]);
+
+    const didDocument = await operationProcessor.resolve(didUniqueSuffix);
+
+    expect(didDocument).toBeDefined();
+    const publicKey2 = Document.getPublicKey(didDocument!, 'key2');
+    expect(publicKey2).toBeDefined();
+    expect(publicKey2!.owner).toBeUndefined(); // if update above went through, the owner would be defined
+  });
+
+  it('should pick earlier of two conflicting updates', async () => {
+    await operationProcessor.processBatch([createOp!]);
+
+    const update1Payload = {
+      didUniqueSuffix,
+      operationNumber: 1,
+      previousOperationHash: createOp!.getOperationHash(),
+      patch: [{
+        op: 'replace',
+        path: '/publicKey/1',
+        value: {
+          id: '#key2',
+          type: 'RsaVerificationKey2018',
+          owner: 'did:sidetree:updateid1',
+          publicKeyPem: process.hrtime() // Some dummy value that's not used.
+        }
+      }]
+    };
+
+    const update2Payload = {
+      didUniqueSuffix,
+      operationNumber: 2,
+      previousOperationHash: createOp!.getOperationHash(),
+      patch: [{
+        op: 'replace',
+        path: '/publicKey/1',
+        value: {
+          id: '#key2',
+          type: 'RsaVerificationKey2018',
+          owner: 'did:sidetree:updateid2',
+          publicKeyPem: process.hrtime() // Some dummy value that's not used.
+        }
+      }]
+    };
+
+    const updateOperation1Buffer = await OperationGenerator.generateUpdateOperationBuffer(update1Payload, '#key1', privateKey);
+    const updateOperation1 = await addBatchFileOfOneOperationToCas(updateOperation1Buffer, cas, 1, 1, 0);
+    await operationProcessor.processBatch([updateOperation1]);
+
```

Comment:
For extra coverage, do you think we can flip the order or processing and process operation 2 first?

Reply:
Sure, makes sense. Took the suggestion.